src/test/java/org/sonarlint/intellij/analysis/SonarLintTaskTest.java

```diff
@@ -108,7 +116,133 @@ public void testTask() {
+    TaskListener listener = mock(TaskListener.class);
+    getProject().getMessageBus().connect(getProject()).subscribe(TaskListener.SONARLINT_TASK_TOPIC, listener);
+    when(sonarLintAnalyzer.analyzeModule(eq(getModule()), eq(files), any(IssueListener.class), any(ProgressMonitor.class)))
+      .thenAnswer((Answer<AnalysisResults>) invocation -> {
+        IssueListener issueListener = invocation.getArgument(2);
+        Issue issue = mock(Issue.class);
+        ClientInputFile clientInputFile = mock(ClientInputFile.class);
+        when(issue.getInputFile()).thenReturn(clientInputFile);
+        when(clientInputFile.getClientObject()).thenReturn(mock(VirtualFile.class));
+        when(clientInputFile.getPath()).thenReturn("path");
+        issueListener.handle(issue);
+        return analysisResults;
+      });
+
+    task.run(progress);
+
+    verify(sonarLintAnalyzer).analyzeModule(eq(getModule()), eq(files), any(IssueListener.class), any(ProgressMonitor.class));
+    verify(issueManagerMock, never()).insertNewIssue(any(), any());
+    verifyNoMoreInteractions(liveIssueBuilder);
+  }
+
+  @Test
+  public void shouldReportIssueForValidFile() throws Exception {
+    Issue issue = mock(Issue.class);
+    ClientInputFile clientInputFile = mock(ClientInputFile.class);
+    when(issue.getInputFile()).thenReturn(clientInputFile);
+    VirtualFile virtualFile = mock(VirtualFile.class);
+    LiveIssue liveIssue = mock(LiveIssue.class);
+    when(virtualFile.isValid()).thenReturn(true);
+    when(clientInputFile.getClientObject()).thenReturn(virtualFile);
+    when(clientInputFile.getPath()).thenReturn("path");
+    when(liveIssueBuilder.buildLiveIssue(any(), any())).thenReturn(liveIssue);
+
+    TaskListener listener = mock(TaskListener.class);
+    getProject().getMessageBus().connect(getProject()).subscribe(TaskListener.SONARLINT_TASK_TOPIC, listener);
+    when(sonarLintAnalyzer.analyzeModule(eq(getModule()), eq(files), any(IssueListener.class), any(ProgressMonitor.class)))
+      .thenAnswer((Answer<AnalysisResults>) invocation -> {
+        IssueListener issueListener = invocation.getArgument(2);
+        issueListener.handle(issue);
+        return analysisResults;
+      });
+
+    task.run(progress);
+
+    verify(sonarLintAnalyzer).analyzeModule(eq(getModule()), eq(files), any(IssueListener.class), any(ProgressMonitor.class));
+    verify(liveIssueBuilder).buildLiveIssue(any(), any());
+    verify(issueManagerMock).insertNewIssue(virtualFile, liveIssue);
+  }
+
+  @Test
+  public void shouldSkipIfFailedToFindIssueLocation() throws Exception {
+    Issue issue = mock(Issue.class);
+    ClientInputFile clientInputFile = mock(ClientInputFile.class);
+    when(issue.getInputFile()).thenReturn(clientInputFile);
+    VirtualFile virtualFile = mock(VirtualFile.class);
+    when(virtualFile.isValid()).thenReturn(true);
+    when(clientInputFile.getClientObject()).thenReturn(virtualFile);
+    when(clientInputFile.getPath()).thenReturn("path");
```

Comment:
Maybe all this code can be extracted in a common utility method named `buildValidIssue`